tika-parsers/src/main/java/org/apache/tika/parser/hwp/HwpTextExtractorV5.java

```diff
@@ -0,0 +1,520 @@
+import org.apache.poi.poifs.filesystem.DocumentEntry;
+import org.apache.poi.poifs.filesystem.DocumentInputStream;
+import org.apache.poi.poifs.filesystem.Entry;
+import org.apache.poi.poifs.filesystem.POIFSFileSystem;
+import org.apache.poi.util.LittleEndian;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.metadata.Office;
+import org.apache.tika.metadata.OfficeOpenXMLCore;
+import org.apache.tika.metadata.TikaCoreProperties;
+import org.apache.tika.sax.XHTMLContentHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+public class HwpTextExtractorV5 {
+	protected static Logger log = LoggerFactory
+			.getLogger(HwpTextExtractorV5.class);
+
+	private static final byte[] HWP_V5_SIGNATURE = "HWP Document File"
+			.getBytes();
+
+	private static final int[] HWP_CONTROL_CHARS = new int[] { 0, 10, 13, 24,
+			25, 26, 27, 28, 29, 30, 31 };
+	private static final int[] HWP_INLINE_CHARS = new int[] { 4, 5, 6, 7, 8, 9,
+			19, 20 };
+	private static final int[] HWP_EXTENDED_CHARS = new int[] { 1, 2, 3, 11,
+			12, 14, 15, 16, 17, 18, 21, 22, 23 };
+
+	private static final int HWPTAG_BEGIN = 0x010;
+
+	private POIFSFileSystem fs;
+
+	/**
+	 * extract Text from HWP Stream.
+	 * 
+	 * @param source
+	 * @param writer
+	 * @return
+	 * @throws FileNotFoundException
+	 * @throws IOException
+	 * @throws SAXException 
+	 */
+	public void extract(InputStream source, Metadata metadata, XHTMLContentHandler xhtml)
+			throws FileNotFoundException, IOException,
+			UnsupportedHwpFormatException, SAXException {
+		if (source == null || xhtml == null)
+			throw new IllegalArgumentException();
+
+		try {
+			fs = new POIFSFileSystem(source);
+
+			DirectoryNode root = fs.getRoot();
+
+			extract0(root, metadata, xhtml);
+
+		} catch (IOException e) {
+			throw new UnsupportedHwpFormatException(
+					"error occurred when parsing HWP Format, It may not HWP Format.", e);
+		} 
+	}
+
+	private void extract0(DirectoryNode root, Metadata metadata, XHTMLContentHandler xhtml)
+			throws IOException, UnsupportedHwpFormatException, SAXException {
+
+		FileHeader header;
+		try {
+			Entry headerEntry = root.getEntry("FileHeader");
+			
+			if (!headerEntry.isDocumentEntry())
+				throw new UnsupportedHwpFormatException(
+						"cannot parse the File Header");
+
+			header = getHeader(headerEntry);
+			
+			parseSummaryInformation(root, metadata);
+			
+		} catch (IOException e) {
+			throw new UnsupportedHwpFormatException(
+					"cannot parse the File Header", e);
+		}
+
+		if (header == null)
+			throw new UnsupportedHwpFormatException(
+					"cannot parse the File Header");
+
+		if (header.viewtext) {
+			parseViewText(header, root, xhtml);
+		} else {
+			parseBodyText(header, root, xhtml);
+		}
+	}
+
+	private void parseSummaryInformation(DirectoryNode root, Metadata metadata) throws UnsupportedHwpFormatException {
+
+		try {
+			
+			Entry summaryEntry = root.getEntry("HwpSummaryInformation");

```

Comment:
Can you modify this to "\u0000HwpSummaryInformation" or whatever the correct unicode code is for the first character?